### PR TITLE
Add UI metrics proxy

### PR DESCRIPTION
**Pull request by banks**
This PR builds on #8743 and adds a proxy endpoint to allow the UI to request metrics from a configured backend.

I spent a while trying to make sure this was safe against path traversals, before realising in the tests that path traversals that would be dangerous are impossible currently since they'd already have been handled by the API mux before they get to the new proxy handler anyway.

I left the code and tests in though both to be warned/prepared in case that behaviour ever changes and to document the risk.

### CODE REVIEW

File: agent/ui_endpoint.go [hunk 1]
```diff
@@ -379,3 +384,75 @@ func (s *HTTPHandlers) UIGatewayIntentions(resp http.ResponseWriter, req *http.R
...
+	// Parse it into a new URL
+	u, err := url.Parse(newURL)
+	if err != nil {
+		log.Error("couldn't parse target URL", "base_url", cfg.BaseURL, "path", subPath)
+		return nil, BadRequestError{Reason: "Invalid path."}
+	}
+
+	// Clean the new URL path to prevent path traversal attacks and remove any
+	// double slashes etc.
+	u.Path = path.Clean(u.Path)
```
Comment (rboyer) [inline agent/ui_endpoint.go:426]: Alternatively you could do something like:
```
if clean := path.Clean(u.Path); clean != u.Path {
  // burn it with fire
}
```

**Comment by banks** [comment]
# TODO:

 - [ ] Add tests for (and functionality for) adding the headers per the config.

File: agent/ui_endpoint.go (hunk 1 above)
Comment (rboyer) [reply agent/ui_endpoint.go:448]: Well the suggestion I had forces you to _only_ provide cleaned paths as input otherwise it fails

**Review by rboyer — APPROVED** [review]
LGTM

**Pull request merged**